client/my-sites/comments/comment-list/index.jsx

```diff
@@ -206,6 +207,29 @@ export class CommentList extends Component {
 		this.props.undoBulkStatus( selectedComments );
 	}
 
+	getPlaceholderOrEmptyContent() {
+		const { isLoading, comments, siteId } = this.props;
+		const [ emptyMessageTitle, emptyMessageLine ] = this.getEmptyMessage();
+		const zeroComments = size( comments ) <= 0;
+		if ( ( ! siteId || isLoading ) && zeroComments ) {
+			return (
+				<CommentDetailPlaceholder key="comment-detail-placeholder" />
+			);
+		}
+		if ( zeroComments ) {
+			return (
+				<EmptyContent
+					illustration="/calypso/images/comments/illustration_comments_gray.svg"
+					illustrationWidth={ 150 }
+					key="comment-list-empty"
+					line={ emptyMessageLine }
+					title={ emptyMessageTitle }
+				/>
+			);
+		}
+		return null;
+	}
```

Comment:
It was originally much longer with the transition group. I'd prefer to keep it here to avoid rendering falsey values. If we're rather particular about keeping it inline, I can create two bools, showPlaceholder, and showEmptyContent that can be passed through connect.